File: Packs/FiltersAndTransformers/Scripts/ExtractEmailTransformer/ExtractEmailTransformer.py
import demistomock as demisto
from CommonServerPython import *


def main():
    try:
        return_results(extract(argToList(demisto.args().get('value'))))
    except Exception as error:
        return_error(str(error), error)


def extract(values: list[str]) -> list[str]:
    list_results = []
    for val in values:
        list_results.extend(re.findall(emailRegex, (val or '').lower()))
    return list_results


if __name__ in ('__main__', '__builtin__', 'builtins'):
    main()

File: Packs/FiltersAndTransformers/Scripts/ExtractEmailTransformer/ExtractEmailTransformer.js
function extract(values) {
    // Extracts email addresses from a list of values.
    // values: Array of strings.

    const listResults = [];

    for (const val of values) {
        listResults.push(...((val || '').toLowerCase().match(emailRegex) || []));
    }

    return listResults;
}

try {
    const values = argToList(args.value);
    const results = extract(values);
    return results;
} catch (error) {
    throw 'Error occurred while running the script:\n' + error;
}

